### DISCUSSION TITLE
How do I style ruby text?

### DISCUSSION THREAD

**Discussion by konomu** [koreader/koreader#12377] [Q&A]
I'm trying to style the furigana above ruby text so that it isn't so close to the word. This problem arises when I set the font size larger and the space between the lines narrower. I have tried adding a simple style tweak 

```
rt {
    position: relative !important;
    line-height: 1;
    font-size: 50%;
    bottom: 0.5em !important;
}
```

in an attempt to push the text up a little.

The line-height and font-size properties work, but I can't seem to get the position of the rt text to change. It seems to work in browser, but has no effect in koreader.

**Comment by poire-z** [comment]
We don't support `position:relative`.

And our ruby support is a bit of a hack (internally translating all that to inline-tables, with internal rubyBox wrapping elements, that may need styling too).
Anyway, there's still some styling tweak supported.
Our default styling is at:
https://github.com/koreader/crengine/blob/287bf21847f37fc4a43e8cc0937376e5d5e87724/cr3gui/data/html5.css#L240-L257
The latest discussion about these choices is in half of the posts in #8063 - where you will find some screenshots of what we got with different CSS. May be you'll find out what you need to tweak to get what you want.
Initially discussed and tested in https://github.com/koreader/koreader/issues/6223#issuecomment-644110399.

**Comment by konomu** [comment]
I'm able to change some line-heights to get space between rt and rb, but it seems that most configurations will cause the space between lines with ruby text to be larger than lines without ruby text. It would be ideal to have the same space between lines to be the same whether ruby text is present or not. There is default tweak in the ruby settings that claims to do this with 

```
p, li { line-height: 2 !important; }
```

but it doesn't work when rt font size is set larger to 50% (which is also available as a default tweak), or when the line spacing is set below a certain threshold. 

I wasn't able to figure out a way to getting the rt properly spaced away from rb without ending up with inconsistent line spacing while keeping the larger font size, but I managed to aleviate the inconsistent line spacing with 

```
rt {
  font-size: 50%;
  line-height: 1;
}

p, li { line-height: 1.5 !important; }

rb { line-height: 1; }
```
It technically doesn't completely solve the inconsistent spacing issue at low line spacing, but now it is much less noticeable. And with this configuration, if line spacing is set above 133%, there is no inconsistent line spacing.
